sources/parsers/src/Evaluator.cpp

```diff
@@ -0,0 +1,202 @@
+#define yyFlexLexer DecimalFlexLexer
+#include <FlexLexer.h>
+#include "DecimalScanner.h"
+
+#undef yyFlexLexer
+/// Float lexer definition.
+#define yyFlexLexer FloatFlexLexer
+#include <FlexLexer.h>
+#include "FloatScanner.h"
+
+#undef yyFlexLexer
+/// String lexer definition.
+#define yyFlexLexer StringFlexLexer
+#include <FlexLexer.h>
+#include "StringScanner.h"
+
+PARSER::Evaluator::Evaluator(const std::string &rSourceName) :
+    m_rSourceName(rSourceName),
+    m_Context(nullptr, 0)
+{
+}
+
+void PARSER::Evaluator::ParseDecimalString(const std::string &input)
+{
+    std::istringstream iss(input);
+    if (!ParseDecimalStream(iss))
+    {
+        throw PARSER::EXCEPTIONS::ParsingError(ParsingErrorMessage(input));
+    }
+    m_Expression = input;
+}
+
+void PARSER::Evaluator::ParseFloatString(const std::string &input)
+{
+    std::istringstream iss(input);
+    if (!ParseFloatStream(iss))
+    {
+        throw PARSER::EXCEPTIONS::ParsingError(ParsingErrorMessage(input));
+    }
+    m_Expression = input;
+}
+
+void PARSER::Evaluator::ParseStringString(const std::string &input)
+{
+    std::istringstream iss(input);
+    if (!ParseStringStream(iss))
+    {
+        throw PARSER::EXCEPTIONS::ParsingError(ParsingErrorMessage(input));
+    }
+    m_Expression = input;
+}
+
+void PARSER::Evaluator::SetContext(const std::smatch *pGroups, uint64_t lineNo) const
+{
+    m_Context.Set(pGroups, lineNo);
+}
+
+PARSER::ExpressionContext &PARSER::Evaluator::GetContext() const
+{
+    return m_Context;
+}
+
+std::tuple<double, std::string> PARSER::Evaluator::EvaluateDouble() const
+{
+    double value;
+    std::string stringValue;
+    try
+    {
+        std::tie(value, stringValue) = m_Context.GetExpression()->EvaluateDouble();
+    }
+    catch (const EXCEPTIONS::EvaluatorException &evaluatorError)
+    {
+        throw PARSER::EXCEPTIONS::EvaluatorException(EvaluationErrorMessage(evaluatorError.what()));
+    }
+
+    if (stringValue.empty())
+    {
+        std::ostringstream strStream;
+        strStream << value;
+        stringValue = strStream.str();
+    }
+
+    return std::make_tuple(value, stringValue);
+}
+
+uint64_t PARSER::Evaluator::EvaluateUint() const
+{
+    try
+    {
+        return m_Context.GetExpression()->EvaluateUint().GetValue();
+    }
+    catch (const PARSER::EXCEPTIONS::EvaluatorException &evaluatorError)
+    {
+        throw PARSER::EXCEPTIONS::EvaluatorException(EvaluationErrorMessage(evaluatorError.what()));
+    }
+}
+
+std::string PARSER::Evaluator::EvaluateString() const
+{
+    try
+    {
+        return m_Context.GetExpression()->EvaluateString();
+    }
+    catch (const PARSER::EXCEPTIONS::EvaluatorException &evaluatorError)
+    {
+        throw PARSER::EXCEPTIONS::EvaluatorException(EvaluationErrorMessage(evaluatorError.what()));
+    }
+}
+
+bool PARSER::Evaluator::ParseDecimalStream(std::istream &in)
+{
+    DecimalScanner scanner(&in);
+    #if defined(PARSER_DEBUG)
+    scanner.set_debug(false);
```

Comment:
If parser debug defined then set debug to false?
Sounds weird.

Reply:
No doubts here! It was weird and got fixed.